typing/ctype.ml

```diff
@@ -2087,11 +2097,18 @@ let unbox_once env ty =
     begin match Env.find_type p env with
     | exception Not_found -> Missing p
     | decl ->
+      let apply ty2 = apply env decl.type_params ty2 args in
       begin match find_unboxed_type decl with
-      | None -> Final_result
       | Some ty2 ->
         let ty2 = match get_desc ty2 with Tpoly (t, _) -> t | _ -> ty2 in
-        Stepped (apply env decl.type_params ty2 args)
+        Stepped (apply ty2)
+      | None -> begin match decl.type_kind with
+        | Type_record_unboxed_product ([{ld_type = arg; _}], Record_unboxed_product) ->
+          Stepped (apply arg)
+        | Type_record_unboxed_product ((_::_::_ as lbls), Record_unboxed_product) ->
+          Stepped_record_unboxed_product (List.map (fun ld -> apply ld.ld_type) lbls)
+        | _ -> Final_result
```

Comment:
Sorry, I meant list the cases for the `_` on the previous line (so all the `type_kind`s).  It looks like instead you've listed all the type_descs a few lines down (which is probably not necessary, since this is really just looking for a couple specific ones.